modules/core/src/v2/coins/eos.ts

```diff
@@ -1067,10 +1069,14 @@ export class Eos extends BaseCoin {
       const txJsonFromHexHeaders = _.pick(txJsonFromHex, eosTransactionHeaderFields);
       const txJsonFromPackedTrxHeaders = _.pick(txJsonFromPackedTrx, eosTransactionHeaderFields);
 
-      // milliseconds are dropped in packed_trx and txHex
+      // dates are rounded to the nearest second in packed_trx and txHex
       _.map([txJsonFromPackedTrxHeaders, txJsonFromHexHeaders, txPrebuild.headers], (headers) => {
         const date = moment(headers.expiration as string);
-        headers.expiration = date.milliseconds(0).toISOString();
+
+        headers.expiration = date
```

Comment:
Please refer to the logic of setting the expiration header in platform:

https://github.com/BitGo/bitgo-microservices/blob/develop/packages/wallet-platform/app/controllers/api/v2/coins/eos.js#L1748

The two shall be similar.